### PR TITLE
HDDS-5666. Add option to createKey via streaming api in Freon

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
Add an option to createKey via streaming api in Freon via OzoneClientKeyGenerator. 

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-5666

## How was this patch tested?
Tested the command via docker

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java [hunk 1]
```diff
@@ -93,6 +103,9 @@ public Void call() throws Exception {
 
     contentGenerator = new ContentGenerator(keySize, bufferSize);
     metadata = new HashMap<>();
+    replicationConfig = ReplicationConfig
```
Comment (captainzmc) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java:106]: This line will conflict with [the patch of EC](https://github.com/apache/ozone/pull/2996/files#diff-28399319a85aa26781a7ab36e4ae8232b086656016e76071edad5ed8b3b6430eR117). EC is currently voting on merge to master, so EC will merge into master before streaming.  So can we continue this PR after EC merge?

**Review by captainzmc — COMMENTED** [review]
Thanks @sadanand48 work on this. I left few comments.

**Comment by szetszwo** [comment]
@sadanand48 , there are conflicts.  Could could you update the change?

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java [hunk 2]
```diff
@@ -78,11 +81,18 @@
   )
   private String omServiceID = null;
 
+  @Option(
+      names = {"--enable_streaming", "--stream"},
```
[the remark this answers is not in the document]
Comment (sadanand48) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java]: Done.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java (hunk 1 above)
Comment (sadanand48) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java:106]: Sure.

Comment (szetszwo) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java:106]: @sadanand48 , let's hardcode replicationConfig in createStreamKey(..) for the moment.  We may change it later if there is a need.
```
  private void createStreamKey(long counter) throws Exception {
    final ReplicationConfig replicationConfig = ReplicationConfig
        .fromProtoTypeAndFactor(HddsProtos.ReplicationType.RATIS,
            HddsProtos.ReplicationFactor.THREE);
    final String key = generateObjectName(counter);

    timer.time(() -> {
      try (OzoneDataStreamOutput stream = bucket
          .createStreamKey(key, keySize, replicationConfig, metadata)) {
        contentGenerator.write(stream);
      }
      return null;
    });
  }
```

**Pull request merged**